openshift/buildConfig-template.yaml

```diff
@@ -4,37 +4,92 @@ metadata:
   name: solver-buildconfig
   annotations:
     description: >
-      This is Thoth Solver Fedora 26+ BuildConfig, this template is meant to
+      This is Thoth Solver Fedora 27+ BuildConfig, this template is meant to
       be used by Bots, but could also be used by humans...
-    openshift.io/display-name: 'Thoth Core: Solver Fedora 26+ BuildConfig'
-    version: 0.2.0
-    tags: poc,thoth,solver,ai-stacks
+    openshift.io/display-name: "Thoth Core: Solver Fedora 26+ BuildConfig"
+    version: 0.6.0
+    tags: thoth,solver,ai-stacks,aistacks
     template.openshift.io/documentation-url: https://github.com/Thoth-Station/
     template.openshift.io/long-description: >
-      This is Thoth Solver Fedora 26+ BuildConfig, this template is meant to
+      This is Thoth Solver Fedora 27+ BuildConfig, this template is meant to
       be used by Bots, but could also be used by humans...
     template.openshift.io/provider-display-name: Red Hat, Inc.
   labels:
     template: solver-buildconfig
     app: thoth
     component: solver
 
+parameters:
+  - description: Name of the github repository
+    displayName: Git Repository
+    required: true
+    name: GITHUB_URL
+    value: "https://github.com/thoth-station/solver"
+
+  - description: Git reference to be used
+    displayName: Git Reference
+    required: true
+    name: GITHUB_REF
+    value: "master"
+
+  - description: Tag of the output ImageStream the resulting container image should go to
+    displayName: ImageStream Tag
+    required: true
+    name: IMAGE_STREAM_TAG
+    value: "latest"
+
 objects:
   - kind: BuildConfig
     apiVersion: v1
     metadata:
-      name: solver-f27
+      name: solver-f29
+      labels:
+        app: thoth
+        service: solver
```

Comment:
Service?

Reply:
"solver-name" is an alternative to "service" I have chosen service because I thought that this is part of a microservice that is doing the solving...